pkg/apis/cluster/v1alpha1/machineclass_types.go

```diff
@@ -41,8 +42,7 @@ type MachineClass struct {
 	// this field is consistent with the underlying machine that will
 	// be provisioned when this class is used, to inform higher level
 	// automation (e.g. the cluster autoscaler).
-	// TODO(hardikdr) Add allocatable field once requirements are clear from autoscaler-clusterapi // integration topic.
-	// Capacity corev1.ResourceList `json:"capacity"`
+	Capacity corev1.ResourceList `json:"capacity"`
```

Comment:
These fields seem very error prone, I'm wondering if it would make more sense to defer the setting of these fields to the provider implementation.

I've also been thinking about how we currently use MachineClass and wonder if it would make more sense to limit it to just MachineDeployments/MachineSets.

/cc @vincepri 
Tagging in Vince, since this relates very much to the Data Model discussions underway.